src/views/Login.js

```diff
@@ -1,7 +1,110 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { withRouter, Link } from 'react-router-dom'
+import styled from 'styled-components'
+
+import mutate from '../utils/mutate'
+import graphql from 'babel-plugin-relay/macro'
+
+import { Form, Container } from 'react-bootstrap'
+import { Text, Button, Layout } from '../styles'
+
+const mutation = graphql`
+  mutation LoginMutation($input: ObtainJSONWebTokenInput!){
+    tokenAuth(input: $input) {
+      token
+    }
+  } 
+`
+
+const Signup = ({ history, props }) => {
+  const [data, setData] = useState([])
+  const { username, password } = data
+  const [error, setError] = useState('none')
+
+  const validateForm = () => {
+    return data.password && data.username
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    const variables = {
+      input: {
+        username: username,
+        password: password
+      }
+    }
+
+    mutate(mutation, variables)
+      .then(e => {
+        const { token } = e.tokenAuth
+        localStorage.setItem('token', token)
```

Comment:
This method should only ever appear in the provider. The provider is your centralized location for all auth logic. You should instead use the react hook `useContext` to import the provider correctly, then use the provider's methods to log in. This will ensure that the provider has the correct current value and can provide it throughout the application.